service/src/test/java/com/myhome/services/unit/AuthenticationSDJpaServiceTest.java

```diff
@@ -0,0 +1,131 @@
+
+public class AuthenticationSDJpaServiceTest {
+
+  private final String USER_ID = "test-user-id";
+  private final String USERNAME = "test-user-name";
+  private final String USER_EMAIL = "test-user-email";
+  private final String USER_PASSWORD = "test-user-password";
+  private final String REQUEST_PASSWORD = "test-request-password";
+  private final Duration TOKEN_LIFETIME = Duration.ofDays(1);
+  private final String SECRET = "secret";
+
+  @Mock
+  private UserSDJpaService userSDJpaService;
+  @Mock
+  private AppJwtEncoderDecoder appJwtEncoderDecoder;
+  @Mock
+  private PasswordEncoder passwordEncoder;
+  @InjectMocks
+  private AuthenticationSDJpaService authenticationSDJpaService;
+
+  @BeforeEach
+  void setUp() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  void loginSuccess(){
+    // given
+    LoginRequest request = getDefaultLoginRequest();
+    UserDto userDto = getDefaultUserDtoRequest();
+    AppJwt appJwt = getDefaultJwtToken(userDto);
+    String encodedJwt = appJwtEncoderDecoder.encode(appJwt, SECRET);
+    ReflectionTestUtils.setField(authenticationSDJpaService, "tokenExpirationTime", Duration.ofDays(1));
+    given(userSDJpaService.findUserByEmail(request.getEmail()))
+        .willReturn(Optional.of(userDto));
+    given(passwordEncoder.matches(request.getPassword(), userDto.getEncryptedPassword()))
+        .willReturn(true);
+    given(appJwtEncoderDecoder.encode(appJwt, SECRET))
+        .willReturn(encodedJwt);
+
+    // when
+    AuthenticationData authenticationData = authenticationSDJpaService.login(request);
+
+    // then
+    assertNotNull(authenticationData);
+    assertEquals(authenticationData.getUserId(), userDto.getUserId());
+    assertEquals(authenticationData.getJwtToken(), encodedJwt);
+    verify(userSDJpaService).findUserByEmail(request.getEmail());
+    verify(passwordEncoder).matches(request.getPassword(), userDto.getEncryptedPassword());
+    verify(appJwtEncoderDecoder).encode(appJwt, SECRET);
+  }
+
+  @Test
+  void loginUserNotFound(){
+    // given
+    LoginRequest request = getDefaultLoginRequest();
+    given(userSDJpaService.findUserByEmail(request.getEmail()))
+        .willReturn(Optional.empty());
+
+    // when and then
+    Assertions.assertThrows(UserNotFoundException.class, () -> authenticationSDJpaService.login(request));
+  }
+
+  @Test
+  void loginCredentialsAreIncorrect(){
+    // given
+    LoginRequest request = getDefaultLoginRequest();
+    UserDto userDto = getDefaultUserDtoRequest();
+    given(userSDJpaService.findUserByEmail(request.getEmail()))
+        .willReturn(Optional.of(userDto));
+    given(passwordEncoder.matches(request.getPassword(), userDto.getEncryptedPassword()))
+        .willReturn(false);
+
+    // when and then
+    Assertions.assertThrows(CredentialsIncorrectException.class, () -> authenticationSDJpaService.login(request));
+
+  }
+  private LoginRequest getDefaultLoginRequest() {
+    LoginRequest request = new LoginRequest();
```

Comment:
Generated classes (in this case `LoginRequest`) have something similar to builder pattern.
You can write `LoginRequest.email(USER_EMAIL).password(REQUEST_PASSWORD)`